--- a/src/util.cc
+++ b/src/util.cc
@@ -1228,6 +1228,10 @@
                                   const char** charsetp, size_t* charsetlenp,
                                   const char* in, size_t len, bool defaultUTF8)
 {
+  if (len>0 && in[len-1]==';'){
+    --len;
+  }
   const char *p = in, *eop = in + len, *mark_first = nullptr,
              *mark_last = nullptr;
   int state = CD_BEFORE_DISPOSITION_TYPE;
